### PR TITLE
BERT doc fixes and usability enhancement

### PR DESCRIPTION
**Pull request by eric-haibin-lin**
## Description ##
- Fixed a few documentation errors
- Fixed a ctx usage in BERT fine-tune script when GPU is not enabled.

@cgraywang @fierceX could you help review? 

### Essentials ###
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

### Changes ###
- [x] Tested in test_bert_models

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: scripts/bert/finetune_classifier.py [hunk 1]
```diff
@@ -69,7 +69,7 @@
 test_batch_size = args.test_batch_size
 lr = args.lr
 
-ctx = mx.cpu() if args.gpu is None or args.gpu == '' else mx.gpu()
+ctx = mx.cpu() if not args.gpu else mx.gpu()
 
 bert, vocabulary = bert_12_768_12(dataset_name='book_corpus_wiki_en_uncased',
```
Comment (cgraywang) [inline scripts/bert/finetune_classifier.py]: Would it be more flexiable to make `dataset_name` as an argument and `assert` when the `dataset_name` and `do_lower_case` doesn't align? e.g., dataset_name is 'uncased' but `do_lower_case=False` as below:

```
bert, vocabulary = bert_12_768_12(dataset_name='book_corpus_wiki_en_uncased',
                                  pretrained=True, ctx=ctx, use_pooler=True,
                                  use_decoder=False, use_classifier=False)
tokenizer = FullTokenizer(vocabulary, do_lower_case=False)
```

File: docs/api/modules/model.rst [hunk 2]
```diff
@@ -48,6 +48,8 @@ Bidirectional Encoder Representations from Transformers
     BERTEncoder
     BERTEncoderCell
     BERTPositionwiseFFN
+    bert_12_768_12
+    bert_24_1024_16
```
Comment (cgraywang) [inline docs/api/modules/model.rst]: Why we didn't add [BERT-Large, Cased](https://storage.googleapis.com/bert_models/2018_10_18/cased_L-24_H-1024_A-16.zip) this time?

Comment (eric-haibin-lin) [reply docs/api/modules/model.rst]: There are a few more models in the pipeline and I'll add them in the next PR

**Comment by mli** [comment]
Job PR-444/1 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-444/1/index.html

File: scripts/bert/finetune_classifier.py (hunk 1 above)
Comment (cgraywang) [reply scripts/bert/finetune_classifier.py]: Sounds good me.

**Comment by mli** [comment]
Job PR-444/2 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-444/2/index.html

**Pull request merged**